src/System.Device.Gpio/System/Device/Gpio/Drivers/SysFsDriver.Linux.cs

```diff
@@ -15,13 +15,33 @@ namespace System.Device.Gpio.Drivers
     public class SysFsDriver : UnixDriver
     {
         private const string GpioBasePath = "/sys/class/gpio";
+        private const string GpioChip = "gpiochip";
+        private const string GpioLabel = "/label";
+        private const string GpioContoller = "pinctrl";
+        private const string GpioOffsetBase = "/base";
         private int _pollFileDescriptor = -1;
         private Thread _eventDetectionThread;
         private int _pinsToDetectEventsCount;
         private readonly static CancellationTokenSource s_eventThreadCancellationTokenSource = new CancellationTokenSource();
         private readonly List<int> _exportedPins = new List<int>();
         private readonly Dictionary<int, UnixDriverDevicePin> _devicePins = new Dictionary<int, UnixDriverDevicePin>();
         private readonly int _pollingTimeoutInMilliseconds = Convert.ToInt32(TimeSpan.FromMilliseconds(1).TotalMilliseconds);
+        private int _pinOffset = 0;
+
+        public SysFsDriver() {
+            IEnumerable<string> fileNames = Directory.EnumerateFileSystemEntries(GpioBasePath);
+            foreach (string name in fileNames)
+            {
+                if (name.Contains(GpioChip))
+                {
+                    if (File.ReadAllText($"{name}{GpioLabel}").StartsWith(GpioContoller))
+                    {
+                        _pinOffset = int.Parse(File.ReadAllText($"{name}{GpioOffsetBase}"));
```

Comment:
>_pinOffset [](start = 24, length = 10)

we are reading the same value all the time. I think it would be better to convert this to a static value that you read it once.

Reply:
it is reading once in constuctor, will make it static.